### PR TITLE
add Shards field in DNStore

### PR DESCRIPTION
**Pull request by w-zr**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

Rename `CNNode`/`DNNode`/`LogNode` to `CNStore`/`DNStore`/`LogStore`.

Add `Shards` field in `DNStore`.

### CODE REVIEW

**Comment by lni** [comment]
The change looks good to me, just be aware that in v0.6, there should be 0 or 1 DN shard in each DNStore. In future releases, that might change.

**Review by lni — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by fengttt — APPROVED** [review]
Sorry I missed the discussion.   What is the purpose of this one?

Anyway, I trust you guys on this.

File: pkg/vm/engine/txn/engine.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (e *Engine) Nodes() (engine.Nodes, error) {
 	}
 
 	var nodes engine.Nodes
-	for _, node := range clusterDetails.CNNodes {
+	for _, node := range clusterDetails.CNStores {
```
[the remark this answers is not in the document]
Comment (w-zr) [reply pkg/vm/engine/txn/engine.go]: done.

File: pkg/vm/engine/txn/shard.go [hunk 2]
```diff
@@ -24,9 +24,9 @@ import (
 type Shard = metadata.DNShard
 
 type ShardPolicy interface {
-	Vector(vec *vector.Vector, nodes []logservicepb.DNNode) ([]*ShardedVector, error)
-	Batch(batch *batch.Batch, nodes []logservicepb.DNNode) ([]*ShardedBatch, error)
-	Nodes(nodes []logservicepb.DNNode) ([]Shard, error)
+	Vector(vec *vector.Vector, nodes []logservicepb.DNStore) ([]*ShardedVector, error)
+	Batch(batch *batch.Batch, nodes []logservicepb.DNStore) ([]*ShardedBatch, error)
+	Nodes(nodes []logservicepb.DNStore) ([]Shard, error)
```
[the remark this answers is not in the document]
Comment (w-zr) [reply pkg/vm/engine/txn/shard.go]: done

**Comment by reusee** [comment]
@fengttt It's for doing sharding in the engine. The distributed sender routes requests by their target shard.
https://github.com/matrixorigin/matrixone/blob/main/pkg/vm/engine/txn/shard.go#L26

**Pull request merged**